app/components/withSelectDir/withSelectDir.js

```diff
@@ -0,0 +1,26 @@
+import { remote } from 'electron'
+
+/**
+ * Invokes electron open dialog
+ */
+
+/**
+ *
+ * @param {string} mode electron dialog.showOpenDialog compatible mode
+ */
+function openDialog(mode = 'openFile') {
+  const { dialog } = remote
+  const [path] = dialog.showOpenDialog({
+    properties: [mode]
+  })
+
+  return path
+}
+
+/**
+ * Render prop that allows children to call native select directory dialog
+ * @param {*} param0
+ */
+const withSelectDir = ({ render }) => render({ openDialog })
```

Comment:
Should either be named `withSelectFile` (since the default is to select a file), or better still `withOpenDialog` since this HOC adds the ability to use openDialog in modes other than just file selection mode.

Reply:
Yep agree. I've extended the render prop scope but haven't updated the name